src/OmniSharp.Abstractions/Models/v1/Completion/CompletionItem.cs

```diff
@@ -0,0 +1,177 @@
+        /// </summary>
+        [JsonProperty("detail")]
+        public string? Detail { get; set; }
+
+        /// <summary>
+        /// A human-readable string that represents a doc-comment. This is
+        /// formatted as markdown.
+        /// </summary>
+        [JsonProperty("documentation")]
+        public string? Documentation { get; set; }
+
+        /// <summary>
+        /// Select this item when showing.
+        /// </summary>
+        [JsonProperty("preselect")]
+        public bool Preselect { get; set; }
+
+        /// <summary>
+        /// A string that should be used when comparing this item
+        /// with other items. When null or empty the label is used.
+        /// </summary>
+        [JsonProperty("sortText")]
+        public string? SortText { get; set; }
+
+        /// <summary>
+        /// A string that should be used when filtering a set of
+        /// completion items. When null or empty the label is used.
+        /// </summary>
+        [JsonProperty("filterText")]
+        public string? FilterText { get; set; }
+
+        /// <summary>
+        /// A string that should be inserted into a document when selecting
+        /// this completion.When null or empty the label is used.
+        /// </summary>
+        [JsonProperty("insertText")]
+        public string? InsertText { get; set; }
+
+        /// <summary>
+        /// The format of <see cref="InsertText"/>.
+        /// </summary>
+        [JsonProperty("insertTextFormat")]
+        public InsertTextFormat? InsertTextFormat { get; set; }
+
+        /// <summary>
+        /// An optional set of characters that when pressed while this completion is active will accept it first and
+        /// then type that character.
+        /// </summary>
+        [JsonProperty("commitCharacters")]
+        public ImmutableArray<char>? CommitCharacters { get; set; }
+
+        /// <summary>
+        /// An optional array of additional text edits that are applied when
+        /// selecting this completion.Edits must not overlap (including the same insert position)
+        /// with the main edit nor with themselves.
+        ///
+        /// Additional text edits should be used to change text unrelated to the current cursor position
+        /// (for example adding an import statement at the top of the file if the completion item will
+        /// insert an unqualified type).
+        /// </summary>
+        [JsonProperty("additionalTextEdits")]
+        public ImmutableArray<TextEdit>? AdditionalTextEdits { get; set; }
+
+        /// <summary>
+        /// Index in the completions list that this completion occurred.
+        /// </summary>
+        [JsonProperty("data")]
+        public int Data { get; set; }
+
+        public override string ToString()
+        {
+            return $"{{ {nameof(Label)} = {Label}, {nameof(CompletionItemKind)} = {Kind} }}";
+        }
+    }
+
+    public struct TextEdit
```

Comment:
I went and implemented new versions of these to line up with the lsp definitions here: it made my vscode development experience simpler. The reason is that `Point` and `Range` as they exist in omnisharp today are 1-based and pascalCased, whereas the lsp definitions are 0-based and camelCased. I need to take these and do a real mapping on the vscode side and get rid of them here, but I haven't yet.